core/src/main/java/org/apache/accumulo/core/spi/balancer/TableLoadBalancer.java

```diff
@@ -106,6 +123,41 @@ protected TabletBalancer getBalancerForTable(TableId tableId) {
     return balancer;
   }
 
+  private SortedMap<TabletServerId,TServerStatus> getCurrentSetForTable(
+      SortedMap<TabletServerId,TServerStatus> allTServers,
+      Map<String,Set<TabletServerId>> groupedTServers, String resourceGroup) {
+
+    String groupName =
```

Comment:
I don't think we should throw an exception when there are zero tablet servers, this could be temporary situation that happens normally.  Maybe log a warning and do nothing.

Also I don't think we should fall back to some default when a user configured group has no tservers in it, could log a warning and take no action.

Reply:
It would be good to have ITs that test zero tservers in a group case.  Maybe something like the following.

 1. Create a table that requires its tablet be assigned to tservers in group X. Tserver group X currently has not tservers.
 2. Start a thread that tries to write to the above table.
 3. Start a tserver in group X.
 4. Verify the write in the thread was successful.

Reply:
I added this test in db29813